sdk/framework/backend.go

```diff
@@ -247,13 +248,18 @@ func (b *Backend) HandleRequest(ctx context.Context, req *logical.Request) (*log
 		}
 	}
 
+	// We need to check SQLConnectionProducer fields separately since they are not top-level Path fields.
+	var sqlFields map[string]any
+	if req.MountType == "database" {
```

Comment:
should we also limit this to requests to the `/config` path? Otherwise I think this will call reflect for every request to a DB path. We probably want to avoid that for hot paths like `/creds` etc.